addons/movments_accounts_balances/controllers/controllers.py

```diff
@@ -79,3 +97,183 @@ def general_ledger_report(self, account_id, start_date, end_date):
+        for line in bill_vals:
+            invoice_line_vals = {
+                'name': line.get('description', ''),
+                'price_unit': line.get('amount', 0.0),
+                'account_id': line.get('account_id'),
+                'analytic_account_id': line.get('analytic_account_id', False),
+            }
+            invoice_lines.append((0, 0, invoice_line_vals))
+
+        # Create a new bill record
+        bill = self.env['account.move'].create({
+
+            'move_type': 'in_invoice',
+            'partner_id': partner_id,
+            'invoice_date': invoice_date,
+            'invoice_date_due': invoice_date_due,
+            'ref': ref,
+            'narration': narration,
+            'invoice_line_ids': invoice_lines,
+        })
+
+        return bill
+
+
+
+    @api.model
+    def unpost_entry(self, entry):
+        if entry.state != 'posted':
+            return f"Journal entry (ID: {entry.id}) is not posted, no action required."
+
+        try:
+            # Unpost the entry
+            entry.button_cancel()
+
+            # You can now make changes to the entry or its related items
+            # ...
+
+            return f"Journal entry (ID: {entry.id}) unposted successfully."
+        except Exception as e:
+            return f"Failed to unpost the journal entry (ID: {entry.id}): {str(e)}"
+
+
+
+    @api.model
+    def delete_bills(self, start_date, end_date):
+        # Define the domain to select records for deletion
+        domain = [('date', '>=', start_date),
+                  ('date', '<=', end_date),
+                  ('move_type', '=', 'in_invoice')]
+
+        # Search for records in the 'account.move' model based on the domain
+        bills_to_delete = self.env['account.move'].search(domain)
+
+        deleted_entry_ids = []
+
+        for bill in bills_to_delete:
+            # Unreconcile the bill
+            if bill.line_ids:
+                for line in bill.line_ids:
+                    if line.reconciled:
+                        line.remove_move_reconcile()
+
+            try:
+                # Unpost the entry
+                bill.button_draft()
+            except Exception as e:
+                return f"Failed to unpost the journal entry (ID: {bill.id}): {str(e)}"
+
+            try:
+                # Delete the entry
+                bill.unlink()
+            except Exception as e:
+                return f"Failed to delete the journal entry (ID: {bill.id}): {str(e)}"
+
+            deleted_entry_ids.append(bill.id)
+
+        return f"{len(deleted_entry_ids)} bills un-reconciled and deleted successfully."
+
+    @api.model
+    def delete_bill(self, bill_id):
+        bill = self.env['account.move'].browse(bill_id)
+        deleted_entry_ids = []
+
+        if bill:
+            # Unreconcile the bill
+            if bill.line_ids:
+                for line in bill.line_ids:
+                    if line.reconciled:
+                        line.remove_move_reconcile()
```

Comment:
I don't think this is needed. Assume there will be no reconciliation in Odoo. Try this code:
```
    @api.model
    def reset_and_delete_vendor_bill(self, journal_entry_id):
        Bill = self.env['account.move']
        bill = Bill.search([('id', '=', journal_entry_id), ('move_type', '=', 'in_invoice')])

        if not bill:
            return "Bill not found."

        # Check if the bill is posted (validated)
        if bill.state == 'posted':
            # Add logic to cancel related journal entries (if any)
            # Example: bill.button_draft() or bill.button_cancel()
            try:
                bill.button_draft()  # Reset to draft
            except Exception as e:
                return "Failed to reset bill to draft: {}".format(e)

        # Additional logic to unreconcile payments if the bill is reconciled

        try:
            bill.unlink()  # Delete the bill
            return "Bill deleted successfully."
        except Exception as e:
            return "Failed to delete bill: {}".format(e)